monkey/monkey_island/cc/services/infection_lifecycle.py

```diff
@@ -14,7 +10,7 @@ def is_report_done(agent_repository: IAgentRepository) -> bool:
     infection_done = _is_infection_done(agent_repository)
 
     if infection_done:
-        _on_finished_infection()
+        ReportService.get_report()
```

Comment:
The name of this method is a bit confusing. This method should be called `update_report` IMO, since it sets `ReportService._report`. Getting the report could be done with getter named `get_report`, that would make more sense, but not required for this PR